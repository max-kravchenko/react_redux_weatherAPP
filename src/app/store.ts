import { configureStore, combineReducers } from '@reduxjs/toolkit';
import weatherSliceReducer from '../features/weatherSlice';

const rootReducer = combineReducers({
  weatherSliceReducer,
});

export const store = configureStore({
  reducer: rootReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
export type AppStore = typeof store; 
export type AppDispatch = AppStore['dispatch'];
