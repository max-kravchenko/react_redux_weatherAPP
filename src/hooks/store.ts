import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import { AppDispatch, RootState } from '../app/store';

export const useCustomDispatch = () => useDispatch<AppDispatch>();

export const useCustomSelector: TypedUseSelectorHook<RootState> = useSelector;