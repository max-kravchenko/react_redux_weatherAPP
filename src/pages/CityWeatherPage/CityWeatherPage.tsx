import { Card, CardMedia, CardContent, Typography } from '@mui/material';
import React from 'react';
import { useCustomSelector } from '../../hooks/store';
import { selectCurrentWeatherData } from '../../selectors/selector';
import Box from '@mui/material/Box';

export const CityWeatherPage = React.memo(function CityCityWeatherPage() {

  const currentCity = window.location.hash.split('/')[1];

  console.log(currentCity);

  const { weather, isLoading, error } = useCustomSelector(selectCurrentWeatherData);

  const filteredWeather = weather.filter((city) => city.name === currentCity.split('%20').join(' '));
  
  return (
    <Card>
      <Typography variant="h3" paddingTop={4}>
        Weather in {filteredWeather[0].name} today
      </Typography>
      <CardContent>
        <Box
          maxWidth={200}
          display="flex"
          justifyContent='space-between'
          alignItems='center'
        >
          <CardMedia
            component="img"
            image={`http://openweathermap.org/img/wn/${filteredWeather[0].weather[0].icon}.png`}
            alt="weather logo"
          />
          <Box
            display="flex"
            alignItems='center'
            justifyContent='center'
            flexDirection='column'
          >
            <Typography variant="h5" >
              {filteredWeather[0].weather[0].main}
            </Typography>
            <Typography variant="h5" >
              {Math.floor(filteredWeather[0].main.temp)}°C
            </Typography>
          </Box> 
        </Box>
        <Typography gutterBottom variant="body1" component="div">
          The current weather in {filteredWeather[0].name} can be described as following: it is {Math.floor(filteredWeather[0].main.temp)}°C with {filteredWeather[0].weather[0].description} but feels like {Math.floor(filteredWeather[0].main.feels_like)}°C with humidity of {filteredWeather[0].main.humidity}% and pressure of {filteredWeather[0].main.pressure} atm.
        </Typography>
        <Typography gutterBottom variant="body1" component="div">
          The speed of wind is {filteredWeather[0].wind.speed} km / hour.
        </Typography>
        <Typography>
          Highs today are around {Math.floor(filteredWeather[0].main.temp_max)}°C, while lows are at {Math.floor(filteredWeather[0].main.temp_min)}°C.
        </Typography>
        <Typography variant="body2" color="text.secondary">
          {filteredWeather[0].sys.country}, {filteredWeather[0].name}
        </Typography>
      </CardContent>
    </Card>
  );
});