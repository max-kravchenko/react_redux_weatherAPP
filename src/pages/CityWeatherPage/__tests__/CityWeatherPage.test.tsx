import React from 'react';
import { Provider } from 'react-redux';
import { store } from '../../../app/store';
import { CityWeatherPage } from '../CityWeatherPage';

describe('CityWeatherPage', () => {
  it('should render correctly', () => {
    expect(<Provider store={store}> <CityWeatherPage/></Provider>).toMatchSnapshot();
  });
});
