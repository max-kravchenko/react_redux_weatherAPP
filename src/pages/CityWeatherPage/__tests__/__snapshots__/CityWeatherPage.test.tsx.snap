// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityWeatherPage should render correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
   
  <Memo(CityCityWeatherPage) />
</Provider>
`;
